packages/tabs/src/tab-nav.vue

```diff
@@ -43,16 +43,25 @@
 
     computed: {
       navStyle() {
-        const dir = ['top', 'bottom'].indexOf(this.rootTabs.tabPosition) !== -1 ? 'X' : 'Y';
+        const dir = this.isHorizontal ? 'X' : 'Y';
         return {
           transform: `translate${dir}(-${this.navOffset}px)`
         };
       },
       sizeName() {
-        return ['top', 'bottom'].indexOf(this.rootTabs.tabPosition) !== -1 ? 'width' : 'height';
+        return this.isHorizontal ? 'width' : 'height';
+      },
+      isHorizontal() {
+        return ['top', 'bottom'].indexOf(this.rootTabs.tabPosition) !== -1;
+      }
+    },
+    watch: {
```

Comment:
seems we don't need a watcher here

Reply:
without these codes can't working if you switch `bottom -> top -> right` in this [case](https://jsfiddle.net/otzLqu50/)

Reply:
I tried `bottom -> top -> right` works fine without this watcher. Any reproduction link, please?

Reply:
should click `tab11` first.  reprod link: https://jsfiddle.net/xu3jqc6w/